### PR TITLE
Decoderlargefiles

### PR DESCRIPTION
**Pull request by singh96aman**
Issue - https://github.com/baler-collaboration/baler/issues/266

Uncommented Decoder batched [code]
Using Numpy Matrix Multiplication instead of List-Wise Multiplication that is creating Memory Issues.

Attached are plots. No impact
[comparison.pdf](https://github.com/baler-collaboration/baler/files/11647855/comparison.pdf)

### CODE REVIEW

File: baler/modules/data_processing.py [hunk 1]
```diff
@@ -158,10 +160,11 @@ def renormalize_func(norm_data: ndarray, min_list: List, range_list: List) -> nd
         ndarray: Array with the un-normalized values.
     """
     norm_data = np.array(norm_data)
-    renormalized = [
-        renormalize_std(norm_data, min_list[i], range_list[i])
```
Comment (singh96aman) [inline baler/modules/data_processing.py:162]: This code is converting the input matrix of 1000x24 size to 1000x24x1000 size, thereby creating large memory issues I believe.

File: baler/modules/data_processing.py [hunk 2]
```diff
@@ -158,10 +160,6 @@ def renormalize_func(norm_data: ndarray, min_list: List, range_list: List) -> nd
...
-        for i in range(len(min_list))
-    ]
-    renormalized_full = [(renormalized[i][:, i]) for i in range(len(renormalized))]
-    renormalized_full = np.array(renormalized_full).T
-    return renormalized_full
+    min_list = np.array(min_list)
+    range_list = np.array(range_list)
+    return norm_data * range_list + min_list
```
Comment (singh96aman) [inline baler/modules/data_processing.py:165]: Numpy Matrix Multiplication and Addition are much more Optimized than ListWise Multiplication.
After testing on CPU/GPU, it seems to be faster and consumes less memory than older code.

File: baler/modules/helper.py [hunk 3]
```diff
@@ -562,27 +562,24 @@ def decompress(model_path, input_path, model_name, config):
...
+            if idx == 0:
+                decompressed = out
+            else:
+                decompressed = np.concatenate((decompressed, out))
 
     if config.data_dimension == 2 and config.model_type == "dense":
         decompressed = decompressed.view(
             len(decompressed), number_of_columns, number_of_columns
         )
-    decompressed = decompressed.detach().cpu().numpy()
```
Comment (singh96aman) [inline baler/modules/helper.py:584]: No Need as L573 invokes the detacher

File: baler/modules/helper.py [hunk 4]
```diff
@@ -562,27 +562,24 @@ def decompress(model_path, input_path, model_name, config):
...
+        for idx, data_batch in enumerate(tqdm(data_dl)):
+            data_batch = data_batch.to(device)
+
+            out = model.decode(data_batch).to(device)
+            # Converting back to numpyarray
+            out = detacher(out)
+            if idx == 0:
+                decompressed = out
+            else:
+                decompressed = np.concatenate((decompressed, out))
```
Comment (singh96aman) [inline baler/modules/helper.py:578]: Tested on CPU/GPU and the above code works fine

File: baler/modules/data_processing.py (hunk 2 above)
Comment (fritjof-b) [reply baler/modules/data_processing.py:165]: Yes I agree, and the operations looks a lot cleaner as well :-)

**Review by fritjof-b — APPROVED** [review]

**Pull request merged**